tests/src/unit-udt_macro.cpp

```diff
@@ -468,247 +455,248 @@ class derived_person_only_serialize_private : person_without_default_constructor
+    using T = typename Pair::second_type;
+    constexpr bool is_ordered = std::is_same<Json, nlohmann::ordered_json>::value;
+
     SECTION("derived person with default values")
     {
         // serialization of default constructed object
-        T p0;
-        CHECK(json(p0).dump() == "{\"age\":0,\"hair_color\":\"blue\",\"metadata\":null,\"name\":\"\"}");
+        const T p0{};
+        CHECK(Json(p0).dump() == (is_ordered ?
+                                  R"({"age":0,"name":"","metadata":null,"hair_color":"blue"})" :
+                                  R"({"age":0,"hair_color":"blue","metadata":null,"name":""})"));
 
         // serialization
         T p1("Erik", 1, {{"haircuts", 2}}, "red");
-        CHECK(json(p1).dump() == "{\"age\":1,\"hair_color\":\"red\",\"metadata\":{\"haircuts\":2},\"name\":\"Erik\"}");
+        CHECK(Json(p1).dump() == (is_ordered ?
+                                  R"({"age":1,"name":"Erik","metadata":{"haircuts":2},"hair_color":"red"})" :
+                                  R"({"age":1,"hair_color":"red","metadata":{"haircuts":2},"name":"Erik"})"));
 
         // deserialization
-        auto p2 = json(p1).get<T>();
+        auto p2 = Json(p1).template get<T>();
         CHECK(p2 == p1);
 
         // roundtrip
-        CHECK(T(json(p1)) == p1);
-        CHECK(json(T(json(p1))) == json(p1));
+        CHECK(T(Json(p1)) == p1);
+        CHECK(Json(T(Json(p1))) == Json(p1));
 
         // check default value in case of missing field
-        json j = json(p1);
+        Json j = Json(p1);
         j.erase("name");
         j.erase("age");
         j.erase("metadata");
         j.erase("hair_color");
-        T p3 = j.get<T>();
+        const T p3 = j.template get<T>();
         CHECK(p3.getName() == "");
         CHECK(p3.getAge() == 0);
         CHECK(p3.getMetadata() == nullptr);
         CHECK(p3.getHairColor() == "blue");
     }
 }
 
-TEST_CASE_TEMPLATE("Serialization/deserialization of classes with 26 public/private member variables via NLOHMANN_DEFINE_TYPE_INTRUSIVE and NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE", T, // NOLINT(readability-math-missing-parentheses)
-                   persons::person_with_private_alphabet,
-                   persons::person_with_public_alphabet)
+TEST_CASE_TEMPLATE("Serialization/deserialization of classes with 26 public/private member variables via NLOHMANN_DEFINE_TYPE_INTRUSIVE and NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE", Pair, // NOLINT(readability-math-missing-parentheses)
+                   std::pair<nlohmann::json, persons::person_with_private_alphabet>,
+                   std::pair<nlohmann::json, persons::person_with_public_alphabet>,
+                   std::pair<nlohmann::ordered_json, persons::person_with_private_alphabet>,
+                   std::pair<nlohmann::ordered_json, persons::person_with_public_alphabet>)
 {
+    using Json = typename Pair::first_type;
+    using T = typename Pair::second_type;
+
     SECTION("alphabet")
     {
-        {
```

Comment:
Out of curiosity: why did you remove the tests below?

Reply:
Because of this review comment on a similar PR: https://github.com/nlohmann/json/pull/2843#discussion_r910626988